data/src/main/java/com/microsoft/azure/kusto/data/auth/CloudInfo.java

```diff
@@ -114,54 +110,59 @@ public static CloudInfo retrieveCloudInfoForCluster(String clusterUrl,
                 }
                 return null;
             });
-
         }
     }
 
     private static CloudInfo fetchImpl(String clusterUrl, @Nullable HttpClient givenHttpClient) throws URISyntaxException, IOException, DataServiceException {
         CloudInfo result;
         HttpClient localHttpClient = givenHttpClient == null ? HttpClientFactory.create(null) : givenHttpClient;
         try {
-            HttpGet request = new HttpGet(UriUtils.appendPathToUri(clusterUrl, METADATA_ENDPOINT));
-            request.addHeader(HttpHeaders.ACCEPT_ENCODING, "gzip,deflate");
-            request.addHeader(HttpHeaders.ACCEPT, "application/json");
+            HttpRequest request = new HttpRequest(HttpMethod.GET, UriUtils.appendPathToUri(clusterUrl, METADATA_ENDPOINT));
```

Comment:
We talked about which client we should use as the default.
Seems like out of our options, OKHttp has a huge advantage - it can automatically decompress "gzip", saving us this code.

It doesn't support "deflate" by default, but this implementation doesn't seem to either - so we should remove it from the accept encoding anyway.

If we do keep this though, it should really be somewhere where the logic can be re-used

Reply:
It should be easy to swap in OkHttp at this point. All that should be needed now is to edit the pom file to exclude azure-core-http-netty and then add the azure-core-http-okhttp package. 👍

Reply:
Removed deflate in [this commit.](https://github.com/Azure/azure-kusto-java/pull/335/commits/9370530077b2a12aa51279d3b7f4f491b63ef6f7)